From 73a265935de2aade2ecffad829aded9b14f99e01 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 16 Oct 2012 16:03:07 +0800
Subject: [PATCH 075/602] ENGR00229800 Update board config to align Android JB
 build env

Remove unused property setting in init.freescale.rc
Update PRODUCT_AAPT_CONFIG for different screen support based on board config
Update /system/etc/permisssion based on board's config
Enable flag dalvik.gc.type-precise for dalvik
Define PRODUCT_CHARACTERISTICS to be tablet

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 arm2_6sl/init.rc                     |  3 ---
 imx6/arm2_6dq.mk                     | 22 ++++++++++++++++++++++
 imx6/arm2_6sl.mk                     | 13 +++++++++++++
 imx6/imx6.mk                         |  6 +++---
 imx6/sabreauto_6q.mk                 | 22 ++++++++++++++++++++++
 imx6/sabresd/SabreSDProductCommon.mk | 24 ++++++++++++++++++++++++
 imx6/sabresd/init.rc                 |  8 --------
 7 files changed, 84 insertions(+), 14 deletions(-)

diff --git a/arm2_6sl/init.rc b/arm2_6sl/init.rc
index f6b19eb..abe4fb4 100755
--- a/arm2_6sl/init.rc
+++ b/arm2_6sl/init.rc
@@ -51,9 +51,6 @@ on boot
     # Add support wakeup system from debug UART
     write /sys/devices/platform/imx-uart.0/tty/ttymxc0/power/wakeup enabled
 
-    # Indicate device type is tablet or SystemUI has problem without HW acceleration
-    setprop sys.devicy.type tablet
-
 service wpa_supplicant /system/bin/wpa_supplicant \
     -Dnl80211 -e/data/misc/wifi/entropy.bin
     class late_start
diff --git a/imx6/arm2_6dq.mk b/imx6/arm2_6dq.mk
index a84204f..81e057e 100644
--- a/imx6/arm2_6dq.mk
+++ b/imx6/arm2_6dq.mk
@@ -12,3 +12,25 @@ include device/fsl/imx6/arm2/ProductCommonArm2.mk
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6dq/init.rc:root/init.freescale.rc
 
+
+PRODUCT_CHARACTERISTICS := tablet
+
+PRODUCT_AAPT_CONFIG += large tvdpi hdpi
+
+PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
+	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
+	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
+	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
+	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
+	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
+
+# for PDK build, include only when the dir exists
+# too early to use $(TARGET_BUILD_PDK)
+ifneq ($(wildcard packages/wallpapers/LivePicker),)
+PRODUCT_COPY_FILES += \
+	packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml
+endif
diff --git a/imx6/arm2_6sl.mk b/imx6/arm2_6sl.mk
index fefe643..019d9be 100644
--- a/imx6/arm2_6sl.mk
+++ b/imx6/arm2_6sl.mk
@@ -12,3 +12,16 @@ include device/fsl/imx6/arm2/ProductCommonArm2.mk
 PRODUCT_COPY_FILES +=	\
 	device/fsl/arm2_6sl/init.rc:root/init.freescale.rc \
 	device/fsl/arm2_6sl/mxckpd.kl:system/usr/keylayout/mxckpd.kl
+
+PRODUCT_CHARACTERISTICS := tablet
+
+PRODUCT_AAPT_CONFIG += large hdpi
+
+PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
+	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
+	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
+	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index cf97699..89a8ae5 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -247,7 +247,7 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     com.android.future.usb.accessory
 
-PRODUCT_LOCALES += hdpi mdpi
+PRODUCT_AAPT_CONFIG := normal mdpi
 
 PRODUCT_COPY_FILES +=	\
 	device/fsl/common/input/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
@@ -268,8 +268,8 @@ PRODUCT_COPY_FILES +=	\
 	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
 	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
 
-PRODUCT_COPY_FILES += \
-    frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
+# we have enough storage space to hold precise GC data
+PRODUCT_TAGS += dalvik.gc.type-precise
 
 # for property
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES := \
diff --git a/imx6/sabreauto_6q.mk b/imx6/sabreauto_6q.mk
index 39d4ee1..0ecfda1 100644
--- a/imx6/sabreauto_6q.mk
+++ b/imx6/sabreauto_6q.mk
@@ -18,3 +18,25 @@ PRODUCT_COPY_FILES += \
 # GPU files
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/sabreauto_6q/overlay
+
+PRODUCT_CHARACTERISTICS := tablet
+
+PRODUCT_AAPT_CONFIG += large hdpi
+
+PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
+	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
+	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
+	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
+	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
+	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
+
+# for PDK build, include only when the dir exists
+# too early to use $(TARGET_BUILD_PDK)
+ifneq ($(wildcard packages/wallpapers/LivePicker),)
+PRODUCT_COPY_FILES += \
+	packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml
+endif
diff --git a/imx6/sabresd/SabreSDProductCommon.mk b/imx6/sabresd/SabreSDProductCommon.mk
index 00cb447..fb665f9 100644
--- a/imx6/sabresd/SabreSDProductCommon.mk
+++ b/imx6/sabresd/SabreSDProductCommon.mk
@@ -7,3 +7,27 @@ PRODUCT_COPY_FILES += \
 # GPU files
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/imx6/sabresd/overlay
+
+PRODUCT_CHARACTERISTICS := tablet
+
+PRODUCT_AAPT_CONFIG += large tvdpi hdpi
+
+PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
+	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
+	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
+	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
+	frameworks/native/data/etc/android.hardware.sensor.gyroscope.xml:system/etc/permissions/android.hardware.sensor.gyroscope.xml \
+	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
+	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
+	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
+	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
+	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml
+
+# for PDK build, include only when the dir exists
+# too early to use $(TARGET_BUILD_PDK)
+ifneq ($(wildcard packages/wallpapers/LivePicker),)
+PRODUCT_COPY_FILES += \
+	packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml
+endif
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index e77d869..28f04b2 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -24,9 +24,6 @@ on boot
     setprop hwc.filter_stretch  1
     setprop hwc.enable_dither   1
 
-    # deine board type
-    setprop sys.devicy.type  "tablet"
-
     # fsl omx graphic manager media framework property
     setprop media.omxgm.enable-player 1
     setprop media.omxgm.enable-record 1
@@ -48,11 +45,6 @@ on boot
     # Set rotation to 270 to cofigure as portrait mode
     setprop ro.sf.hwrotation 0
 
-    # hdmi audio output
-    setprop ro.HDMI_AUDIO_OUTPUT 1
-
-    # Set video overlay display mode
-    setprop sys.VIDEO_OVERLAY_DISPLAY 1
 # change for FSL specific service
     chown system system /sys/devices/system/cpu/cpu0
     chown system system /sys/devices/system/cpu/cpu1
-- 
1.8.0

